azure/scope/azure_secret_cloud.go

```diff
@@ -2,43 +2,281 @@ package scope
 
 import (
 	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net/http"
 	"os"
-	"path/filepath"
-	"strings"
+	"sync"
+	"time"
 
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
 	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/rest"
+	azurepkg "sigs.k8s.io/cluster-api-provider-azure/azure"
 	"sigs.k8s.io/cluster-api-provider-azure/util/tele"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 const (
-	AzureEnvironentFolderEnvName = "AZURE_ENVIRONMENT_FOLDER"
+	// Deprecated: Use ConfigMaps instead
+	AzureEnvironentFolderEnvName = "AZURE_ENVIRONMENT_FOLDER_CAPZ"
```

Comment:
What will happen in the cases where AZURE_ENVIRONMENT_FOLDER was used. This no more exists now